### PR TITLE
Refactors Grab Code (and adds Collar Grabbing)

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
You can now grab anyone by their pet collar if you target the head and they are wearing a pet collar! This will deal extremely minor oxygen damage to them.

## Why It's Good For The Game
It's funny. 

## Changelog
:cl:
add: You can now grab people by their pet collar, assuming they have one.
refactor: Grab code
/:cl:

### CODE REVIEW

**Comment by MCterra10** [comment]
speedclose

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -290,18 +290,32 @@
 
 	if(ismob(AM))
 		var/mob/M = AM
+		var/grab_sanity_check = FALSE
 
 		log_combat(src, M, "grabbed", addition="passive grab")
 		if(!supress_message && !(iscarbon(AM) && HAS_TRAIT(src, TRAIT_STRONG_GRABBER))) //Everything in this if statement handles chat messages for grabbing
 			var/mob/living/L = M
-			if (L.getorgan(/obj/item/organ/tail) && zone_selected == BODY_ZONE_PRECISE_GROIN) //Does the target have a tail?
+			if(istype(L, /mob/living/carbon/human))//Everything in this statement handles special grab messages that require checking item slot contents
```
Comment (Archanial) [inline code/modules/mob/living/living.dm]: We have helper for this

Comment (Kapu1178) [reply code/modules/mob/living/living.dm]: fixed in commit

**Comment by ghost** [comment]
Based

**Review by PowerfulBacon — DISMISSED** [review]
This is funny but a cluttered mess in the code

**Comment by PowerfulBacon** [comment]
(Btw oxy damage is menacingly in small amounts since it heals automatically)

**Comment by OceanFish1** [comment]
New achievement added: Kill a person by cuffing them, putting a pet collar on and make them suffocate to death using grab.

**Comment by utilbird** [comment]
is collar grabbing worth having a clothing check for it every single time someone is grabbed

**Comment by Kapu1178** [comment]
> grabbed

image

**Review by ike709 — CHANGES_REQUESTED** [review]
This is bad code. In theory a good way to do this would be to have equipping/removing a pet collar add/remove a "choker" mob trait and checking for that.

**Comment by Kapu1178** [comment]
Will be adding a helper at some point next week and using that instead

**Comment by Kapu1178** [comment]
Helper added. A ton of code uses the istype(H.wear_neck, /obj/item/neck/blahblah)

So once this gets merged ill do a seperate pr to replace most/all of the uses with the new helper.

File: code/modules/mob/inventory.dm [hunk 2]
```diff
@@ -508,3 +508,10 @@
 			SEND_SIGNAL(A, COMSIG_TRY_STORAGE_RETURN_INVENTORY, item_stuff)
 			processing_list += item_stuff
 	return processing_list
+
+//Checks carbon-based wearable item slots.
+/mob/proc/check_wearable_slot(mob/M, item, slot)
+	if(iscarbon(M))
+		if(istype(M.get_item_by_slot(slot), item))
+			return TRUE
+	return FALSE
```
Comment (utilbird) [inline code/modules/mob/inventory.dm]: 99% sure there's already a helper for this. This doesn't improve the code quality much regardless. I'd recommend going with ike's flag suggestion

Comment (Kapu1178) [reply code/modules/mob/inventory.dm]: me and ike talked about it briefly like 2-3 weeks ago and the conclusion was this solution was alright. I really don't like the idea of a flag for such a specific thing.

Comment (utilbird) [reply code/modules/mob/inventory.dm]: Having specific exceptions in fundamental interaction procs is pretty meh. If @ike709's okay with it though then this is fine for merge

**Pull request closed without merging**